python/dgl/heterograph.py

```diff
@@ -4110,6 +4110,32 @@ def is_homograph(self):
         """Return if the graph is homogeneous."""
         return len(self.ntypes) == 1 and len(self.etypes) == 1
 
+    def sparse_format(self, etype=None):
+        """Return sparse format of the given edge/relation type.
+
+        Parameters
+        ----------
+        etype : str, optional
+            The edge/relation type. Can be omitted if there is only
+            one node type in the graph. (Default: None)
+
+        Returns
+        -------
+        string : 'any', 'coo', 'csr', or 'csc'
```

Comment:
Return `any` is quite strange to be honest. Can we have two APIs:
```python
def format(self, etype=None, return_all=False):
    """Return the sparse formats in use of the given edge/relation type.

    Returns
    --------
    string or list of string
        If return_all is True, return all the formats currently in use (could be multiple).
        Otherwise, return any one that is in use.
    """

def restrict_format(self, etype=None):
    """Return the allowed sparse formats of the given edge/relation type.
    """
```
To change the format we have:
```python
def to_format(self, format, etype=None):
    """Return a clone graph but stored in the given sparse format.

    If 'any' is given, the restrict formats of the returned graph is relaxed.

    Parameters
    -------------
    A new graph.
    """
```

Reply:
rename format to format_in_use because format is a reserved word in python class.